sdk-tests/src/test/java/io/dapr/it/testcontainers/workflows/crossapp/WorkflowsCrossAppCallActivityIT.java

```diff
@@ -0,0 +1,267 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import io.dapr.testcontainers.Subscription;
+
+/**
+ * Cross-App Pattern integration test.
+ *
+ * This test demonstrates the cross-app pattern by:
+ * 1. Starting 3 Dapr containers (crossapp-worker, app2, app3)
+ * 2. Launching Java processes that register workflows/activities in separate apps
+ * 3. Executing a cross-app workflow
+ * 4. Asserting successful completion
+ */
+@Testcontainers
+@Tag("testcontainers")
+public class WorkflowsCrossAppCallActivityIT {
+
+  private static final Network DAPR_NETWORK = Network.newNetwork();
+  
+  // Shared placement and scheduler containers for all Dapr instances
+  private static DaprPlacementContainer sharedPlacementContainer;
+  private static DaprSchedulerContainer sharedSchedulerContainer;
+  
+  // Main workflow orchestrator container
+  private static DaprContainer MAIN_WORKFLOW_CONTAINER;
+  
+  // App2 container for App2TransformActivity - using DaprContainer with custom ports
+  private static DaprContainer APP2_CONTAINER;
+  
+  // App3 container for App3FinalizeActivity - using DaprContainer with custom ports
+  private static DaprContainer APP3_CONTAINER;
+
+  // TestContainers for each app
+  private static GenericContainer<?> crossappWorker;
+  private static GenericContainer<?> app2Worker;
+  private static GenericContainer<?> app3Worker;
+
+  @BeforeAll
+  public static void setUp() throws Exception {
+    sharedPlacementContainer = new DaprPlacementContainer("daprio/placement:1.16.0-rc.3")
+        .withNetwork(DAPR_NETWORK)
+        .withNetworkAliases("placement")
+        .withReuse(false);
+    sharedPlacementContainer.start();
+    
+    sharedSchedulerContainer = new DaprSchedulerContainer("daprio/scheduler:1.16.0-rc.3")
+        .withNetwork(DAPR_NETWORK)
+        .withNetworkAliases("scheduler")
+        .withReuse(false);
+    sharedSchedulerContainer.start();
+
+    // Initialize Dapr containers with shared placement and scheduler
+    MAIN_WORKFLOW_CONTAINER = new DaprContainer(DAPR_RUNTIME_IMAGE_TAG)
+        .withAppName("crossapp-worker")
+        .withNetwork(DAPR_NETWORK)
+        .withNetworkAliases("main-workflow-sidecar")
+        .withPlacementContainer(sharedPlacementContainer)
+        .withSchedulerContainer(sharedSchedulerContainer)
+        .withComponent(new Component("kvstore", "state.in-memory", "v1", Map.of("actorStateStore", "true")))
+        .withComponent(new Component("pubsub", "pubsub.in-memory", "v1", Collections.emptyMap()))
+        .withSubscription(new Subscription("local", "pubsub", "topic", "/events"))
+        .withDaprLogLevel(DaprLogLevel.DEBUG)
+        .withLogConsumer(outputFrame -> System.out.println("MAIN_WORKFLOW: " + outputFrame.getUtf8String()))
+        .withAppChannelAddress("host.testcontainers.internal")
+        .waitingFor(Wait.forHttp("/v1.0/healthz/outbound")
```

Comment:
@cicoyle this is already built in in the DaprContainer, if a condition was missing, it will be better to add it to the DaprContainer logic that adding more every time a user creates a new instance.